File: server/src/main/java/com/pass/auth/infra/JwtTokenProperties.java
package com.pass.auth.infra;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "auth.jwt")
public record JwtTokenProperties(
        @NotBlank
        String secretKey,

        @NotNull
        @Positive
        Long expirationTime
) {
}

File: server/src/main/kotlin/com/fluffy/auth/infra/JwtTokenProperties.kt
package com.fluffy.auth.infra

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Positive
import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "auth.jwt")
data class JwtTokenProperties(

    @field:NotBlank
    val secretKey: String,

    @field:NotNull
    @field:Positive
    val expirationTime: Long,
)
